src/Performance/PerformanceConnector.ts

```diff
@@ -0,0 +1,167 @@
+/* *
+ *
+ *  (c) 2009-2025 Highsoft AS
+ *
+ *  License: www.highcharts.com/license
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ *  Authors:
+ *  - Mateusz Bernacik
+ *
+ * */
+
+
+'use strict';
+
+
+/* *
+ *
+ *  Imports
+ *
+ * */
+
+
+import External from '../Shared/External';
+import MorningstarAPI from '../Shared/MorningstarAPI';
+import PAUSConnector from '../Shared/PAUSConnector';
+import MorningstarURL from '../Shared/MorningstarURL';
+import PerformanceOptions, {
+    PerformanceRequestPayload
+} from './PerformanceOptions';
+
+
+/* *
+ *
+ *  Declarations
+ *
+ * */
+
+
+/* *
+ *
+ *  Functions
+ *
+ * */
+
+
+/* *
+ *
+ *  Constants
+ *
+ * */
+
+// export const DATA_TABLES = [
+//     { key: 'TrailingReturns' },
+// ];
+
+
+/* *
+ *
+ *  Class
+ *
+ * */
+
+
+export class PerformanceConnector extends PAUSConnector {
+
+
+    /* *
+     *
+     *  Constructors
+     *
+     * */
+
+
+    public constructor (
+        options: PerformanceOptions
+    ) {
+        super(
+            options
+        );
+
+        this.options = options;
+    }
+
+
+    /* *
+     *
+     *  Properties
+     *
+     * */
+
+    public override readonly options: PerformanceOptions;
+
+    /* *
+     *
+     *  Functions
+     *
+     * */
+
+
+    public override async load (
+        options?: PerformanceOptions
+    ): Promise<any> {
+        await super.load();
+
+        const userOptions = { ...this.options, ...options };
+        const api = this.api = this.api || new MorningstarAPI(userOptions.api);
+        const langcult = userOptions.langcult || 'en-US';
+        const url =
+            new MorningstarURL(`/portfolioanalysis/v1/performance?langcult=${langcult}`, api.baseURL);
+
+        const bodyPayload: PerformanceRequestPayload = {
+            View: {
+                Id: this.options.viewId || 'All'
+            },
+            Config: {
+                Id: this.options.configId || 'QuickPortfolio'
+            },
+            RequestSettings: this.options.requestSettings,
+            Portfolios: this.options.portfolios
+        };
+
+        const response = await api.fetch(url, {
+            body: JSON.stringify(bodyPayload),
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            method: 'POST'
+        });
+
+        const json = await response.json() as unknown;
+
+        // eslint-disable-next-line no-console
+        console.log('Performance', json);
+
+        return this.setModifierOptions(userOptions.dataModifier);
```

Comment:
Q: Could this be the the default method in PAUSConnector? Then we would need to implement only methods like `getViewPayload()`, `getConfigPayload()`, `getRequestSettingsPayload()`, `getPortfoliosPayload()` (maybe without the `Payload` word). Of course the URL could be set in props of Hypo/Perfo/XRay connectors, instead of the hardcoded `"/portfolioanalysis/v1/performance"`.

Reply:
Maybe even simpler: `getPayload()`?